nodes/persistence-node/src/__tests__/jest/integration/ipfs-gateway-api.spec.ts

```diff
@@ -0,0 +1,53 @@
+import { buildMainDependencyContainer } from "../../../modules/daemon/daemon.deps";
+import { Express } from "express";
+import supertest from "supertest";
+
+jest.setTimeout(50000);
+
+describe("ipfs-gateway-api", () => {
+  let app: Express;
+
+  const nonExistentHash = "QmcnrHegojMFqHkRhixazY67Zb9mSbMLv6sSxyDpUtnrQSy";
+  const nonBase32Hash = "SomeInvalidHash";
+
+  beforeAll(async () => {
+    const container = await buildMainDependencyContainer();
+    app = container.cradle.ipfsGatewayApi.createExpressApp();
+  });
+
+  it("cat returns 404 for unknown hash", async () => {
```

Comment:
I thought we settled on following IPFS specs for HTTP API https://docs.ipfs.io/reference/http/api/#http-status-codes 
Unknown has is error 500, with an error in the response
See https://ipfs.io/api/v0/cat?arg=QmP for example

Reply:
I changed it to throw 500 instead of 404